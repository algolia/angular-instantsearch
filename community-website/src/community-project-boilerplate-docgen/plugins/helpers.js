const highlight = require('./syntaxHighlighting.js');
const md = require('./mdRenderer.js');

// this plugin provides ATM one helper to easily compute the publicPath of assets
module.exports = function helpers(filenames, metalsmith, cb) {
  // eslint-disable-next-line no-param-reassign
  metalsmith.metadata().h = {
    markdown(src) {
      return md.render(src);
    },
    highlight(src, opts) {
      const lang = opts && opts.lang;
      return highlight(src, lang);
    },
    maybeActive(navPath, singlePathOrArrayOfPaths) {
      const pathsToTest = [].concat(singlePathOrArrayOfPaths);
      return pathsToTest.some(pathToTest => navPath.indexOf(pathToTest) === 0)
        ? 'active'
        : '';
    },
  };

  cb();
};
