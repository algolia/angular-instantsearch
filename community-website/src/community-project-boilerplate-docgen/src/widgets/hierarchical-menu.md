---
title: Widgets - HierarchicalMenu
layout: widget.pug
---

# Hierarchical Menu

## Description

The hierarchical menu widget is used to create a navigation based on a hierarchy of facet attributes.

It is commonly used for categories with subcategories.

## Options

| Attributes           | Type                | Description
| -                    | -                   | -
| `attributes`         | `string[]`          | Array of attributes to use to generate the hierarchy of the menu
| `limit?`             | `number`            | How much facet values to get
| `separator?`         | `string`            | Separator used in the attributes to separate level values
| `rootPath?`          | `string`            | Prefix path to use if the first level is not the root level
| `showParentLevel?`   | `boolean`           | Show the parent level of the current refined value
| `sortBy?`            | `string[]/Function` | How to sort facet values
| `autoHideContainer?` | `boolean`           | Hides the widget if there's no item to display

## Code example

```html
<ng-ais-hierarchical-menu
  [showParentLevel]="false"
  [attributes]="[
    'hierarchicalCategories.lvl0',
    'hierarchicalCategories.lvl1',
    'hierarchicalCategories.lvl2'
  ]"
>
</ng-ais-hierarchical-menu>
```
