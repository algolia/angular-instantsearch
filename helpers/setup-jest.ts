import "jest-preset-angular";

// stupid workaround for TS
import * as helper from "algoliasearch-helper";
require("algoliasearch-helper").default = helper;

const mock = () => {
  let storage = {};
  return {
    getItem(key) {
      return storage[key] || null;
    },
    setItem(key, value) {
      storage[key] = value;
    },
    removeItem(key) {
      delete storage[key];
    },
    clear() {
      storage = {};
    }
  };
};

Object.defineProperty(window, "localStorage", { value: mock() });
Object.defineProperty(window, "sessionStorage", { value: mock() });
Object.defineProperty(window, "getComputedStyle", {
  value: () => ["-webkit-appearance"]
});
