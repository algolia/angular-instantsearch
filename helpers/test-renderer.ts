import { Component, ViewChild } from "@angular/core";
import { TestBed } from "@angular/core/testing";

import { NgAisInstantSearchModule } from "../src/instantsearch/instantsearch.module";

// mock component, dont create a real instantearch instance
jest.mock("../src/base-widget");
jest.mock("../src/instantsearch/instantsearch");

export function createRenderer({
  template,
  TestedWidget,
  defaultState,
  additionalImports,
  additionalDeclarations
}: {
  template: string;
  TestedWidget: any;
  additionalImports?: any[];
  additionalDeclarations?: any[];
  defaultState?: {};
}) {
  return function(state?: {}, firstRender = false) {
    return render(
      {
        template,
        TestedWidget,
        additionalImports,
        additionalDeclarations,
        state: state ? { ...(defaultState || {}), ...state } : undefined
      },
      firstRender
    );
  };
}

function render(
  {
    template,
    TestedWidget,
    additionalImports,
    additionalDeclarations,
    state
  }: {
    template: string;
    TestedWidget: any;
    additionalImports?: any[];
    additionalDeclarations?: any[];
    state?: {};
  },
  firstRender = false
) {
  @Component({
    template: `
      <ng-ais-instantsearch>
        ${template}
      </ng-ais-instantsearch>
    `
  })
  class TestContainer {
    @ViewChild(TestedWidget) testedWidget;
  }

  TestBed.configureTestingModule({
    declarations: [
      TestContainer,
      TestedWidget,
      ...(additionalDeclarations || [])
    ],
    imports: [NgAisInstantSearchModule.forRoot(), ...(additionalImports || [])]
  });

  const fixture = TestBed.createComponent(TestContainer);
  fixture.detectChanges();

  if (state) {
    fixture.componentInstance.testedWidget.updateState(state, firstRender);
    fixture.detectChanges();
  }

  return fixture;
}
