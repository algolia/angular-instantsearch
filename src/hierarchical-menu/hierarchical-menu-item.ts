import { Component, Input } from '@angular/core';
import { bem } from '../utils';

export type HierarchicalMenuItem = {
  value: string;
  label: string;
  count: number;
  isRefined: boolean;
  data: HierarchicalMenuItem[];
};

@Component({
  selector: 'ais-hierarchical-menu-item',
  template: `
    <li
      [class]="getItemClass(item)"
      (click)="handleClick($event, item)"
    >
      <a
        [class]="cx('link')"
        href="{{createURL(item.value)}}"
        (click)="handleClick($event, item)"
      >
        <span [class]="cx('label')">{{item.label}}</span>
        <span [class]="cx('count')">{{item.count}}</span>
      </a>

      <ul
        [class]="getListClass()"
        *ngIf="item.isRefined && isArray(item.data) && item.data.length > 0"
      >
        <ais-hierarchical-menu-item
          *ngFor="let child of item.data"
          [item]="child"
          [createURL]="createURL"
          [refine]="refine"
          [lvl]="lvl + 1"
        >
        </ais-hierarchical-menu-item>
      </ul>
    </li>
  `,
})
export class NgAisHierarchicalMenuItem {
  @Input() public lvl: number = 1;
  @Input() public refine: (string) => void;
  @Input() public createURL: (string) => string;
  @Input() public item: HierarchicalMenuItem;

  public cx = bem('HierarchicalMenu');

  public getItemClass(item) {
    let className = this.cx('item');

    if (item.isRefined) {
      className = `${className} ${this.cx('item', 'selected')}`;
    }

    if (this.isArray(item.data) && item.data.length > 0) {
      className = `${className} ${this.cx('item', 'parent')}`;
    }

    return className;
  }

  public getListClass() {
    return `${this.cx('list')} ${this.cx('list', 'child')} ${this.cx(
      'list',
      `lvl${this.lvl}`
    )}`;
  }

  public isArray(potentialArray: any) {
    return Array.isArray(potentialArray);
  }

  public handleClick(event: MouseEvent, item: HierarchicalMenuItem) {
    event.preventDefault();
    event.stopPropagation();

    this.refine(item.value);
  }
}
