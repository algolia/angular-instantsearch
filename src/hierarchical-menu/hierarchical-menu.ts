import { Component, Input, Inject, PLATFORM_ID } from "@angular/core";
import { connectHierarchicalMenu } from "instantsearch.js/es/connectors";
import { noop, isFunction } from "lodash-es";

import { BaseWidget } from "../base-widget";
import { NgAisInstance } from "../instantsearch/instantsearch-instance";
import { bem, parseNumberInput } from "../utils";

export type HierarchicalMenuState = {
  createURL: Function;
  items: {}[];
  refine: Function;
};

@Component({
  selector: "ng-ais-hierarchical-menu",
  template: `
    <div [class]="cx()">
      <ng-ais-header [header]="header" [className]="cx('header')"></ng-ais-header>

      <div
        [class]="cx('body')"
        *ngIf="!isHidden"
      >
        <ul [class]="cx('list') + ' ' + cx('list', 'lvl0')">
          <ng-ais-hierarchical-menu-item
            *ngFor="let item of items"
            [item]="item"
            [createURL]="state.createURL"
            [refine]="state.refine"
          >
          </ng-ais-hierarchical-menu-item>
        </ul>
      </div>

      <ng-ais-footer [footer]="footer" [className]="cx('footer')"></ng-ais-footer>
    </div>
  `
})
export class NgAisHierarchicalMenu extends BaseWidget {
  // render option
  @Input() public transformItems?: Function;

  // connector options
  @Input() public attributes: string[];
  @Input() public separator?: string = " > ";
  @Input() public rootPath?: string;
  @Input() public showParentLevel?: boolean;
  @Input() public limit?: number | string = 10;
  @Input() public sortBy?: string[] | ((item: object) => number);

  public state: HierarchicalMenuState = {
    createURL: noop,
    items: [],
    refine: noop
  };

  get isHidden() {
    return this.state.items.length === 0 && this.autoHideContainer;
  }

  get items() {
    return isFunction(this.transformItems)
      ? this.transformItems(this.state.items)
      : this.state.items;
  }

  constructor(
    @Inject(PLATFORM_ID) public platformId: Object,
    searchInstance: NgAisInstance
  ) {
    super(searchInstance, "HierarchicalMenu");
  }

  public ngOnInit() {
    this.createWidget(connectHierarchicalMenu, {
      limit: parseNumberInput(this.limit),
      attributes: this.attributes,
      rootPath: this.rootPath,
      separator: this.separator,
      showParentLevel: this.showParentLevel,
      sortBy: this.sortBy
    });

    super.ngOnInit();
  }
}
