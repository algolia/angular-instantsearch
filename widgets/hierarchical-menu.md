## `<ng-ais-hierarchical-menu />`

The hierarchical menu widget is used to create a navigation based on a hierarchy of facet attributes.

It is commonly used for categories with subcategories.

| Attributes         | Type                | Description
| -                  | -                   | -
| `attributes`       | `string[]`          | Array of attributes to use to generate the hierarchy of the menu
| `limit?`           | `number`            | How much facet values to get
| `separator?`       | `string`            | Separator used in the attributes to separate level values
| `rootPath?`        | `string`            | Prefix path to use if the first level is not the root level
| `showParentLevel?` | `boolean`           | Show the parent level of the current refined value
| `sortBy?`          | `string[]/Function` | How to sort facet values
| `header?`          | `string`            | Displays text before widget
| `footer?`          | `string`            | Displays text after widget