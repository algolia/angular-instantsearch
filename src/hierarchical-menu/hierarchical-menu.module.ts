import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { NgAisHierarchicalMenu } from './hierarchical-menu';
import { NgAisHierarchicalMenuItem } from './hierarchical-menu-item';

@NgModule({
  declarations: [NgAisHierarchicalMenu, NgAisHierarchicalMenuItem],
  entryComponents: [NgAisHierarchicalMenu],
  exports: [NgAisHierarchicalMenu],
  imports: [CommonModule],
})
export class NgAisHierarchicalMenuModule {}
