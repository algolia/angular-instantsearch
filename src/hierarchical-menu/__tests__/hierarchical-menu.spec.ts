import { createRenderer } from '../../../helpers/test-renderer';
import { NgAisHierarchicalMenu } from '../hierarchical-menu';
import { NgAisHierarchicalMenuItem } from '../hierarchical-menu-item';

const defaultState = {
  items: [
    { label: 'one', count: 100, value: 'one' },
    {
      label: 'two',
      count: 100,
      value: 'two',
      isRefined: true,
      data: [
        { label: 'six', count: 100, value: 'six' },
        { label: 'seven', count: 100, value: 'seven' },
        { label: 'eight', count: 100, value: 'eight' },
        { label: 'nine', count: 100, value: 'nine' },
      ],
    },
    { label: 'three', count: 100, value: 'three' },
    { label: 'four', count: 100, value: 'four' },
    { label: 'five', count: 100, value: 'five' },
  ],
  refine: jest.fn(),
  createURL: jest.fn(),
};

const render = createRenderer({
  defaultState,
  template: '<ais-hierarchical-menu></ais-hierarchical-menu>',
  TestedWidget: NgAisHierarchicalMenu,
  additionalDeclarations: [NgAisHierarchicalMenuItem],
});

describe('HierarchicalMenu', () => {
  it('renders markup without state', () => {
    const fixture = render();
    expect(fixture).toMatchSnapshot();
  });

  it('renders markup with state', () => {
    const fixture = render({});
    expect(fixture).toMatchSnapshot();
  });

  it('should call refine() on item click', () => {
    const refine = jest.fn();
    const fixture = render({ refine });

    const [firstItem] = fixture.debugElement.nativeElement.querySelectorAll(
      'li'
    );
    firstItem.click();

    expect(refine).toHaveBeenCalled();
    expect(refine).toHaveBeenCalledWith(defaultState.items[0].value);
  });

  it('should apply `transformItems` if specified', () => {
    const fixture = render({});

    const mapItems = items =>
      items.map(item => ({
        ...item,
        label: `foo - ${item.label}`,
        data: item.data ? mapItems(item.data) : null,
      }));

    fixture.componentInstance.testedWidget.transformItems = mapItems;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be hidden with `autoHideContainer`', () => {
    const fixture = render({ items: [] });
    fixture.componentInstance.testedWidget.autoHideContainer = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
