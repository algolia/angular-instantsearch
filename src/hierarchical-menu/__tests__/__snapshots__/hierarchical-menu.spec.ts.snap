// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HierarchicalMenu renders markup with state 1`] = `
<ng-component
  testedWidget={[Function NgAisHierarchicalMenu]}
>
  
        <ng-ais-instantsearch>
    
          
    <ng-ais-hierarchical-menu>
      
      
      
      <div
        class="ais-HierarchicalMenu"
      >
        
        
        <ul
          class="ais-HierarchicalMenu-list ais-HierarchicalMenu-list--lvl0"
        >
          
          
          
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  one
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item ais-HierarchicalMenu-item--selected ais-HierarchicalMenu-item--parent"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  two
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              <ul
                class="ais-HierarchicalMenu-list ais-HierarchicalMenu-list--child ais-HierarchicalMenu-list--lvl1"
              >
                
          
                
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        six
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        seven
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        eight
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        nine
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                
        
              </ul>
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  three
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  four
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  five
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          
        
        </ul>
        
      
      </div>
      
    
    </ng-ais-hierarchical-menu>
    
        
  </ng-ais-instantsearch>
      
</ng-component>
`;

exports[`HierarchicalMenu renders markup without state 1`] = `
<ng-component
  testedWidget={[Function NgAisHierarchicalMenu]}
>
  
        <ng-ais-instantsearch>
    
          
    <ng-ais-hierarchical-menu>
      
      
      
      <div
        class="ais-HierarchicalMenu"
      >
        
        
        <ul
          class="ais-HierarchicalMenu-list ais-HierarchicalMenu-list--lvl0"
        >
          
          
          
          
        
        </ul>
        
      
      </div>
      
    
    </ng-ais-hierarchical-menu>
    
        
  </ng-ais-instantsearch>
      
</ng-component>
`;

exports[`HierarchicalMenu should apply \`transformItems\` if specified 1`] = `
<ng-component
  testedWidget={[Function NgAisHierarchicalMenu]}
>
  
        <ng-ais-instantsearch>
    
          
    <ng-ais-hierarchical-menu>
      
      
      
      <div
        class="ais-HierarchicalMenu"
      >
        
        
        <ul
          class="ais-HierarchicalMenu-list ais-HierarchicalMenu-list--lvl0"
        >
          
          
          
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  foo - one
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item ais-HierarchicalMenu-item--selected ais-HierarchicalMenu-item--parent"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  foo - two
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              <ul
                class="ais-HierarchicalMenu-list ais-HierarchicalMenu-list--child ais-HierarchicalMenu-list--lvl1"
              >
                
          
                
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        foo - six
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        foo - seven
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        foo - eight
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                <ng-ais-hierarchical-menu-item
                  ng-reflect-create-u-r-l="function () {
          return f"
                  ng-reflect-item="[object Object]"
                  ng-reflect-lvl="2"
                  ng-reflect-refine="function () {
          return f"
                >
                  
      
                  <li
                    class="ais-HierarchicalMenu-item"
                  >
                    
        
                    <a
                      class="ais-HierarchicalMenu-link"
                      href=""
                    >
                      
          
                      <span
                        class="ais-HierarchicalMenu-label"
                      >
                        foo - nine
                      </span>
                      
          
                      <span
                        class="ais-HierarchicalMenu-count"
                      >
                        100
                      </span>
                      
        
                    </a>
                    
  
        
                    
                    
      
                  </li>
                  
    
                </ng-ais-hierarchical-menu-item>
                
        
              </ul>
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  foo - three
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  foo - four
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          <ng-ais-hierarchical-menu-item
            ng-reflect-create-u-r-l="function () {
          return f"
            ng-reflect-item="[object Object]"
            ng-reflect-refine="function () {
          return f"
          >
            
      
            <li
              class="ais-HierarchicalMenu-item"
            >
              
        
              <a
                class="ais-HierarchicalMenu-link"
                href=""
              >
                
          
                <span
                  class="ais-HierarchicalMenu-label"
                >
                  foo - five
                </span>
                
          
                <span
                  class="ais-HierarchicalMenu-count"
                >
                  100
                </span>
                
        
              </a>
              
  
        
              
              
      
            </li>
            
    
          </ng-ais-hierarchical-menu-item>
          
        
        </ul>
        
      
      </div>
      
    
    </ng-ais-hierarchical-menu>
    
        
  </ng-ais-instantsearch>
      
</ng-component>
`;

exports[`HierarchicalMenu should be hidden with \`autoHideContainer\` 1`] = `
<ng-component
  testedWidget={[Function NgAisHierarchicalMenu]}
>
  
        <ng-ais-instantsearch>
    
          
    <ng-ais-hierarchical-menu>
      
      
      
      
    
    </ng-ais-hierarchical-menu>
    
        
  </ng-ais-instantsearch>
      
</ng-component>
`;
